### PR TITLE
Moved xi-editor to Submodule

### PR DESCRIPTION
**Pull request by liamrosenfeld**
This provides:
- One less install step
- Allowing the position of the folder to be moved around and have it definitely recorded in more than the readme
- Guaranteed presence on clone (much harder to mess up install)

While still maintaining the same master tracking master functionality

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,30 +35,40 @@ Screenshot (will need to be updated as syntax coloring and UI polish is added):
...
-> git clone https://github.com/xi-editor/xi-mac
+clone the repository with:
+
+```bash
+> git clone --recurse-submodules https://github.com/xi-editor/xi-mac
 > cd xi-mac
-> git clone https://github.com/xi-editor/xi-editor
+```
+
+and open with
```
Comment (cmyr) [inline README.md]: maybe this should be called 'build and open'? Or we could make these two steps?

File: README.md [hunk 2]
```diff
@@ -35,30 +35,40 @@ Screenshot (will need to be updated as syntax coloring and UI polish is added):
...
 
-
 ## Getting started
 
 ### Requirements
+
 - [Xcode 9.x](https://developer.apple.com/xcode/)
-- [Rust](https://www.rust-lang.org/). We test against the latest stable version,
-and recommend installing through [rustup](https://rustup.rs).
+- [Rust](https://www.rust-lang.org/). We test against the latest stable version, and recommend installing through [rustup](https://rustup.rs).
```
Comment (cmyr) [inline README.md]: total nit, but we manually wrap markdown at either 72 or 80 columns. This makes it render correctly as plaintext, and markdown ignores single newlines when rendering as html.

Comment (liamrosenfeld) [reply README.md]: Reverting that now

File: README.md (hunk 1 above)
Comment (liamrosenfeld) [reply README.md]: That's a much better description. Updating it now.

**Comment by liamrosenfeld** [comment]
@cmyr, I amended the commit to fix both of your reviews.

File: README.md [hunk 3]
```diff
@@ -35,30 +35,41 @@ Screenshot (will need to be updated as syntax coloring and UI polish is added):
...
+and recommend installing through [rustup](https://rustup.rs)
 
+### Installing
 
-Note: the front-end and back-end are split into two separate repositories. This
+*Note:* the front-end and back-end are split into two separate repositories. This
 is the front-end, and the back-end (or core) is now in
-[xi-editor](https://github.com/xi-editor/xi-editor). Make sure to have that checked out
-as a subdirectory.
+[xi-editor](https://github.com/xi-editor/xi-editor). It is contained in a submodule that is checked out during the clone command.
```
Comment (cmyr) [inline README.md]: I think this still needs a line break? 😁

File: README.md [hunk 4]
```diff
@@ -35,30 +35,41 @@ Screenshot (will need to be updated as syntax coloring and UI polish is added):
...
 
-
 ## Getting started
 
 ### Requirements
+
 - [Xcode 9.x](https://developer.apple.com/xcode/)
 - [Rust](https://www.rust-lang.org/). We test against the latest stable version,
-and recommend installing through [rustup](https://rustup.rs).
+and recommend installing through [rustup](https://rustup.rs)
```
Comment (cmyr) [inline README.md]: while i'm nitting, I think this is maybe clearer with the full stop/period?

**Comment by liamrosenfeld** [comment]
@cmyr, amended with the period.

File: README.md (hunk 4 above)
Comment (liamrosenfeld) [reply README.md]: Sorry, I must have missed that when I was coping and pasting from master.

**Comment by liamrosenfeld** [comment]
Thank you!

**Pull request merged**